### ISSUE TITLE
`crln_obs` used instead of `hgln_obs` by `astropy.wcs` but not `sunpy.map`

### ISSUE THREAD

**Issue by svank** [sunpy/sunpy#7184]
### Describe the bug

I *think* this is a bug in Sunpy, but I don't think I understand the full context here.

In short, when parsing the WCS information in a FITS header, information is extracted from the FITS header to define the observer location. When creating a WCS directly, it appears to load the Carrington longitude/latitude (`crln_obs`, `crlt_obs`) and convert it to Stonyhurst coordinates, but when using a Sunpy Map, the Stonyhurst lat/lon in the header are used (`hgln_obs`, `hglt_obs`). The latter seems preferable, since as I understand it the Carrington frame has the most variation in how implementations define it. For at least files where `crln_obs` comes from SunSPICE, there's an offset between its definition and Sunpy's, meaning that the observer location is different depending on whether the file is passed to `WCS` directly or passed though `Map` (with `Map` being correct). Maybe the answer is "just use `Map`", but this seems like an inconsistency worth flagging.

I don't know if this is the right place to fix this (if this is a bug), but I've found that [here](https://github.com/sunpy/sunpy/blob/4343e44ac19c4113c3ba9205d242c5bc2e11e4b2/sunpy/coordinates/wcs_utils.py#L131), if I add a `break` statement at the bottom of this loop, passing the header to `WCS` directly behaves the same as `Map`, using `hgln_obs` if present (because the loop stops when `HeliographicStonyhurst` works). If this seems right, I can make a PR.

In more detail:

I'm working with FITS files from WISPR on PSP.
```python
from astropy.io import fits
import astropy.units as u
from astropy.wcs import WCS
import sunpy.coordinates
from sunpy.map import Map
file = '/home/svankooten/psp/WISPR/WISPR_ENC10_L3_FITS/20211117/psp_L3_wispr_20211117T120315_V1_1221.fits'
header = fits.getheader(file)
wcs = WCS(header)
map = Map(file)
```

The `SkyCoord`s produced by `wcs` and `map` (and `map.wcs`, which does the same as `map`) have different observer longitudes:
```python
In [3]: wcs.pixel_to_world(0,0).observer
Out[3]: 
<HeliographicStonyhurst Coordinate (obstime=2021-11-17T12:03:35.901, rsun=695507.968 km): (lon, lat, radius) in (deg, deg, m)
    (-30.60235774, 2.98625488, 3.0456058e+10)>
In [5]: map.pixel_to_world(0*u.pix, 0*u.pix).observer
Out[5]: 
<HeliographicStonyhurst Coordinate (obstime=2021-11-17T12:03:18.237, rsun=695507.968 km): (lon, lat, radius) in (deg, deg, m)
    (-30.58606346, 2.98625488, 3.0456058e+10)>

In [6]: map.wcs.pixel_to_world(0,0).observer
Out[6]: 
<HeliographicStonyhurst Coordinate (obstime=2021-11-17T12:03:18.237, rsun=695507.968 km): (lon, lat, radius) in (deg, deg, m)
    (-30.58606346, 2.98625488, 3.0456058e+10)>
```

The output from `map` matches the FITS header:
```python
In [7]: header['hgln_obs']
Out[7]: -30.5860634619
```

[This discussion](https://docs.sunpy.org/en/stable/topic_guide/coordinates/carrington.html#compared-to-sunspice) of the difference between Carrington longitudes in Sunpy and SunSPICE applies, as the latter is the source of the WISPR header values. It says Sunpy Carrington longitudes are greater by 0.82° for Earth-bound observers. For this image, PSP is at 0.2 AU, and 0.82° * 0.2 = 0.16°, which is the difference in the two Stonyhurst longitudes, so I think Sunpy is using the Carrington longitude from the FITS file to compute the Stonyhurst observer location, and so this offset from the value in the FITS header is being picked up.

### System Details

```
==============================
sunpy Installation Information
==============================

General
#######
OS: Ubuntu (22.04, Linux 6.2.0-26-generic)
Arch: 64bit, (x86_64)
sunpy: 5.1.dev168+gbadca1c0a.d20230830
Installation path: /home/svankooten/.anaconda/envs/sunpy-dev/lib/python3.11/site-packages/sunpy-5.1.dev168+gbadca1c0a.d20230830.dist-info

Required Dependencies
#####################
astropy: 5.3.2
numpy: 1.24.4
packaging: 23.1
parfive: 2.0.2

Optional Dependencies
#####################
asdf: 2.15.1
asdf-astropy: 0.4.0
beautifulsoup4: 4.12.2
cdflib: 0.0.0
dask: 2023.8.1
drms: 0.6.4
glymur: 0.12.8
h5netcdf: 1.2.0
h5py: 3.9.0
lxml: 4.9.3
matplotlib: 3.7.2
mpl-animators: 1.1.0
pandas: 2.0.3
python-dateutil: 2.8.2
reproject: 0.11.0
scikit-image: 0.21.0
scipy: 1.11.2
sqlalchemy: 2.0.20
tqdm: 4.66.1
zeep: 4.2.1
```

### Installation method

git checkout

**Comment by ayshih** [comment]
> I don't know if this is the right place to fix this (if this is a bug), but I've found that [here](https://github.com/sunpy/sunpy/blob/4343e44ac19c4113c3ba9205d242c5bc2e11e4b2/sunpy/coordinates/wcs_utils.py#L131), if I add a `break` statement at the bottom of this loop, passing the header to `WCS` directly behaves the same as `Map`, using `hgln_obs` if present (because the loop stops when `HeliographicStonyhurst` works). If this seems right, I can make a PR.

Yup, this is a bug, and that's how and where to fix it.  PR away!

**Comment by ayshih** [comment]
For your edification, the reason that `Map` behaves differently is because it creates a new header with WCS information rather than using the original header.  Some map sources have headers that cannot be parsed by `WCS()` without fixing some keyword values.  Some map sources have untrustworthy heliographic coordinates (either type) for the observer location, so the observer location ought to be pulled from other keywords and then transformed to heliographic coordinates.  If you create a `WCS` instance from the original header, you don't get any of this source-specific fixing.

**Comment by svank** [comment]
I was indeed wondering about the difference in behavior, and that's a good reason. Thanks!

**Comment by ayshih** [comment]
Closed by #7188
